spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/editors/LevelEditor.java

```diff
@@ -0,0 +1,16 @@
+package com.baeldung.enummapping.editors;
+
+import java.beans.PropertyEditorSupport;
+
+import org.apache.commons.lang3.EnumUtils;
+
+import com.baeldung.enummapping.enums.Level;
+
+public class LevelEditor extends PropertyEditorSupport {
+
+    @Override
+    public void setAsText(String text) {
+        setValue(EnumUtils.getEnum(Level.class, text.toUpperCase()));
```

Comment:
Shouldn't we check the string is neither null nor empty?

Reply:
Done

Normally, we used _required=false_, so when the parameter isn't specified, the method parameter is bound to null (No converters or property editors are called).

However, I agree. We need to protect our custom converter and property editor against any misuse or NPE

I added additional test cases to take into account _null_ and empty strings